goudpscan/scanner.go

```diff
@@ -96,27 +97,23 @@ func BreakUPIP(segments []string) []string {
 
 		start, err := strconv.Atoi(_start)
 		if err != nil {
-			panic(fmt.Errorf(
-				"Error in breaking up ip ranges: %s",
-				err,
-			))
+			return nil, fmt.Errorf("parsing start int: %w", err)
 		}
 		end, err := strconv.Atoi(_end)
 		if err != nil {
-			panic(fmt.Errorf(
-				"Error in breaking up ip ranges: %s",
-				err,
-			))
+			return nil, fmt.Errorf("parsing end int: %w", err)
 		}
 
 		for i := start; i <= end; i++ {
-			tails = append(tails,
-				BreakUPIP(
-					append(
-						[]string{strconv.Itoa(i)}, segments[idx+1:]...,
-					),
-				)...,
+			r, err := BreakUPIP(
+				append(
+					[]string{strconv.Itoa(i)}, segments[idx+1:]...,
+				),
 			)
+			if err != nil {
+				return nil, fmt.Errorf("breaking UPIP: %w", err)
```

Comment:
Please use something like "failed to break up IP"

Reply:
"failed to" doesn't make make sense in an error message. An error message already indicates failure. What's interesting to know is what you were doing when something went wrong.

Reply:
I see, but there was a failure already and the reason why it happened is following the statement. From this message you also get the same information of what was happening)

Reply:
Whenever you get an `error` in return the failure already happened. There's no need to explain that explicitly.

```
failed to x: failed to y: failed to z: p was wrong
```
vs
```
doing x: doing y: doing z: p was wrong
```
You make your error messages less verbose.

Reply:
Okay, after the discussion we've agreed on the simple form of the error messages, example:
```
read config: open file: file doesn't exist
```